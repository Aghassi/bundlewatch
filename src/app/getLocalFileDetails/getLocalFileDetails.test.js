import getSize from './getSize'
import ValidationError from '../errors/ValidationError'

import getLocalFileDetails from '.'

describe(`getLocalFileDetails`, () => {
    it('Works with basic options', async () => {
        const result = await getLocalFileDetails({
            files: [
                {
                    path: './__testdata__/*.jpg',
                    maxSize: '100kB',
                },
            ],
            defaultCompression: 'none',
        })

        // TODO: assert logger.warn called

        delete result.url
        expect(result).toMatchSnapshot()
    })

    it(`Works when files dont exist, shows warning`, async () => {
        const result = await getLocalFileDetails({
            files: [
                {
                    path: './__testdata__/test-file-doesnt-exist.jpg',
                    maxSize: '100kB',
                },
            ],
        })

        delete result.url
        expect(result).toMatchSnapshot()
    })

    it('Works with CI environment', async () => {
        const MOCK_AUTH_TOKEN = 'mock-auth-token'
        const MOCK_REPO = {
            owner: 'mockowner',
            name: 'mockname',
            currentBranch: 'mockCurrentBranch',
            branchBase: 'mockBranchBase',
            commitSha: 'mockCommitsha',
        }

        const result = await getLocalFileDetails({
            files: [
                {
                    path: './__testdata__/*.jpg',
                    maxSize: '1MB',
                },
            ],
            ci: {
                githubAccessToken: MOCK_AUTH_TOKEN,
                repoOwner: MOCK_REPO.owner,
                repoName: MOCK_REPO.name,
                repoCurrentBranch: MOCK_REPO.currentBranch,
                repoBranchBase: MOCK_REPO.branchBase,
                commitSha: MOCK_REPO.branchBase,
            },
        })

        delete result.url
        expect(result).toMatchSnapshot()
    })
})

describe('getSize', () => {
    it('Throws validations error when using brotli compression without the package', async () => {
        expect(() =>
            getSize({
                filePath: './__testdata__/test-file-1.jpg',
                compression: 'brotli',
            }),
        ).toThrow(ValidationError)
    })
})
