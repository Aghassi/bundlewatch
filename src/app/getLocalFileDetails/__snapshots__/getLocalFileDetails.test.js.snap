// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bundlewatch Node API Works when files dont exist, shows warning 1`] = `
Object {
  "./__testdata__/test-file-doesnt-exist.jpg": Object {
    "error": "There is no matching file for ./__testdata__/test-file-doesnt-exist.jpg",
  },
}
`;

exports[`bundlewatch Node API Works with CI environment 1`] = `
Object {
  "./__testdata__/test-file-1.jpg": Object {
    "compression": undefined,
    "maxSize": 1048576,
    "size": 70145,
  },
  "./__testdata__/test-file-2.jpg": Object {
    "compression": undefined,
    "maxSize": 1048576,
    "size": 458900,
  },
}
`;

exports[`bundlewatch Node API Works with basic options 1`] = `
Object {
  "./__testdata__/test-file-1.jpg": Object {
    "compression": "none",
    "maxSize": 102400,
    "size": 70145,
  },
  "./__testdata__/test-file-2.jpg": Object {
    "compression": "none",
    "maxSize": 102400,
    "size": 458900,
  },
}
`;

exports[`getLocalFileDetails Works when files dont exist, shows warning 1`] = `
Object {
  "./__testdata__/test-file-doesnt-exist.jpg": Object {
    "error": "There is no matching file for ./__testdata__/test-file-doesnt-exist.jpg",
  },
}
`;

exports[`getLocalFileDetails Works with CI environment 1`] = `
Object {
  "./__testdata__/test-file-1.jpg": Object {
    "compression": undefined,
    "maxSize": 1048576,
    "size": 70145,
  },
  "./__testdata__/test-file-2.jpg": Object {
    "compression": undefined,
    "maxSize": 1048576,
    "size": 458900,
  },
}
`;

exports[`getLocalFileDetails Works with basic options 1`] = `
Object {
  "./__testdata__/test-file-1.jpg": Object {
    "compression": "none",
    "maxSize": 102400,
    "size": 70145,
  },
  "./__testdata__/test-file-2.jpg": Object {
    "compression": "none",
    "maxSize": 102400,
    "size": 458900,
  },
}
`;
